qldbdriver/result.go

```diff
@@ -66,9 +68,51 @@ func (result *Result) getNextPage() error {
 	result.pageValues = nextPage.Page.Values
 	result.pageToken = nextPage.Page.NextPageToken
 	result.index = 0
+	result.updateMetrics(nextPage)
 	return nil
 }
 
+func (result *Result) updateMetrics(fetchPageResult *qldbsession.FetchPageResult) {
+	if fetchPageResult.ConsumedIOs != nil {
+		if result.consumedIOs == nil {
+			result.consumedIOs = &IOUsage{new(int64), new(int64)}
+		}
+		*result.consumedIOs.readIOs += *fetchPageResult.ConsumedIOs.ReadIOs
```

Comment:
Can these be nil as well?

Reply:
No, confirmed with Saurabh that this is not possible from SDK.

Reply:
You can't update the result -- it should be considered immutable. Can you refactor this please?

Reply:
The API result object is `fetchPageResult`, not `result` . We take `fetchPageResult.ConsumedIOs.ReadIOs` value and add it to `result.consumedIOs.readIOs`

`result` is an object created in [transaction.go](https://github.com/awslabs/amazon-qldb-driver-go/blob/master/qldbdriver/transaction.go#L73)